src/main/scala/cromwell/engine/workflow/CallMetadataBuilder.scala

```diff
@@ -0,0 +1,190 @@
+        backend = BackendValues.extract(job)
+      } yield key -> baseMetadata.copy(backend = Option(backend.name), jobId = backend.jobId, backendStatus = backend.status)
+    }.toMap
+
+  /**
+   * Function to build a transformer that adds standard streams data to the entries in the input `ExecutionMap`.
+   */
+  private def buildStreamsTransformer(standardStreamsMap: Map[FullyQualifiedName, Seq[StdoutStderr]]): ExecutionMapTransformer =
+    executionMap => {
+      val databaseKeysWithNoneIndexes = executionMap.keys groupBy { _.fqn } filter {
+        case (fqn, edks) => edks.size == 1 && edks.head.index.isEmpty
+      }
+
+      def indexForFqn(fqn: FullyQualifiedName, rawIndex: Int): ExecutionIndex = {
+        if (databaseKeysWithNoneIndexes.contains(fqn)) None else rawIndex.toIndex
+      }
+
+      for {
+        (fqn, seqOfStreams) <- standardStreamsMap.toTraversable
+        (streams, rawIndex) <- seqOfStreams.zipWithIndex
+        key = ExecutionDatabaseKey(fqn, indexForFqn(fqn, rawIndex))
+        baseMetadata = executionMap.get(key).get
+      } yield key -> baseMetadata.copy(streams = Option(streams))
+    }.toMap
+
+  /** Remove symbols corresponding to collectors. */
+  private def filterCollectorSymbols(symbols: Traversable[Symbol]): Traversable[Symbol] = {
+    // Squash duplicate ExecutionDatabaseKeys.
+    val databaseKeys = symbols map { s => ExecutionDatabaseKey(s.scope, s.index.toIndex) } toSet
+    // Look for FQNs with more than one ExecutionDatabaseKey per FQN.
+    val fqnsWithCollectors = databaseKeys.groupBy(_.fqn).collect({ case (fqn, keys) if keys.size > 1 => fqn }).toSet
+    // We know which FQNs with None indexes correspond to collectors, so filter matching symbols.
+    symbols.filterNot { s => s.index.toIndex.isEmpty && fqnsWithCollectors.contains(s.scope) }
+  }
+
+  /**
+   *  Construct the map of `FullyQualifiedName`s to `Seq[CallMetadata]` that contains all the call metadata available
+   *  for the specified parameters.
+   */
+  def build(executions: Traversable[Execution],
+            standardStreamsMap: Map[FullyQualifiedName, Seq[StdoutStderr]],
+            callInputs: Traversable[Symbol],
+            callOutputs: Traversable[Symbol],
+            jobMap: Map[ExecutionDatabaseKey, Any]): Map[FullyQualifiedName, Seq[CallMetadata]] = {
+
+    val executionKeys = executions map { x => ExecutionDatabaseKey(x.callFqn, x.index.toIndex) }
+
+    // Define a sequence of ExecutionMap transformer functions.
+    val executionMapTransformers = Seq(
+      buildBaseTransformer(executions, executionKeys),
+      buildInputsTransformer(callInputs),
+      buildOutputsTransformer(callOutputs),
+      buildJobDataTransformer(executionKeys, jobMap),
+      buildStreamsTransformer(standardStreamsMap)
+    )
+
+    // Fold a zero ExecutionMap across this Seq of functions.
+    val executionMap: ExecutionMap = executionMapTransformers.foldLeft(Map.empty: ExecutionMap) {
```

Comment:
If you have the type ascription in the fold I don't think you need it on the LHS, it seems like it should be easy to ascertain